### PR TITLE
Api resolver setup

### PR DESCRIPTION
**Pull request by ModellbahnFreak**
Implemented the graphQL-Schema as TS-classes for implementing th resolvers

## What does this break
The new schema istn't 100% equal to the current GraphQL schema.
Some properties were renamed and very few were removed/added
__Please check your API usage to make sure your usage is still compatible__

### CODE REVIEW

**Comment by nk-coding** [comment]
CONST!

File: src/api/resolvers/types/mutations/inputs/GraphQLCreateUserInput.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        displayName: {
+            type: GraphQLNonNull(GraphQLString),
+            description: "The name of the user to display in the GUI.\n\nMax. 200 characters."
+        },
+        password: {
+            type: GraphQLNonNull(GraphQLString),
+            description: "The password for the new user in plain text"
+        },
+        email: {
+            type: GraphQLNonNull(GraphQLString),
```
Comment (nk-coding) [inline src/api/resolvers/types/mutations/inputs/GraphQLCreateUserInput.ts]:
```suggestion
            type: GraphQLString,
```

File: src/temp/Element1.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { GraphQLObjectType, GraphQLObjectTypeConfig, GraphQLString } from "graphql";
```
Comment (nk-coding) [inline src/temp/Element1.ts]: remove temporary files?

File: src/api/DBManagerInjector.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * Initializes a new DBManagerInjector. To inject a new Database manager into a request, call `handle(res, req, next)`
+     * 
+     * @param pgClient The postgres database client connected to the database to use for the creation of the database manager.
+     * This can't be `null` or `undefined` and must be a valid postgres client.
+     * @param idGenerator The Snowflake generator to pass to the new database manager for id generation.
+     * This can't be `null` or `undefined` and must be a valid `SnowflakeGenerator`
+     */
+    constructor(pgClient: Client, idGenerator: SnowflakeGenerator) {
```
Comment (nk-coding) [inline src/api/DBManagerInjector.ts]: visibility modifier?

File: src/api/resolvers/CCIMSSchema.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import typeImports from "./typeImports";
+
+let ccimsSchema = new GraphQLSchema({
+    types: typeImports,
+    mutation,
+    query
+});
+export default ccimsSchema;
+
+/*
```
Comment (nk-coding) [inline src/api/resolvers/CCIMSSchema.ts]: why is there a huge amount of commented out code?

File: src/api/resolvers/types/mutations/inputs/GraphQLCreateIMSInput.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            type: GraphQLIMSType,
+            description: "The type/system this IMS is an instance of"
+        },
+        endpoint: {
+            type: GraphQLString,
+            description: `The endpoint where to reach the IMS instance.
+            
+In the most cases this will be a URL in the form of
+\`\`\`
+https://example.com/api/[API_KEY]
```
Comment (nk-coding) [inline src/api/resolvers/types/mutations/inputs/GraphQLCreateIMSInput.ts:23]: sure?
we want to make the changes depending on the user, so there should probably be no api key there

**Review by nk-coding — CHANGES_REQUESTED** [review]

File: src/api/DBManagerInjector.ts (hunk 3 above)
Comment (ModellbahnFreak) [reply src/api/DBManagerInjector.ts]: a882809

File: src/api/resolvers/CCIMSSchema.ts (hunk 4 above)
Comment (ModellbahnFreak) [reply src/api/resolvers/CCIMSSchema.ts]: 70b2ef5

File: src/api/resolvers/types/mutations/inputs/GraphQLCreateIMSInput.ts (hunk 5 above)
Comment (ModellbahnFreak) [reply src/api/resolvers/types/mutations/inputs/GraphQLCreateIMSInput.ts:23]: Maybe bad description. Changed it in 78154e3

**Review by nk-coding — APPROVED** [review]

**Pull request merged**